teammapper-backend/src/map/controllers/maps.gateway.ts

```diff
@@ -176,14 +180,26 @@ export class MapsGateway implements OnGatewayDisconnect {
       return Promise.resolve(false)
 
     const mmpMap: IMmpClientMap = request.map
+
+    // Disconnect all clients temporarily
+    // Emit an event so clients can display a notification
+    this.server.to(mmpMap.uuid).emit('clientNotification', { clientId: client.id, message: 'TOASTS.WARNINGS.MAP_IMPORT_IN_PROGRESS', type: 'warning' })
+
+    this.clients.forEach((client) => client.leave(request.mapId))
```

Comment:
saving all clients in a global map could be problematic:

1) You are iterating over all available clients of all maps, but you are only caring about one map. First optimization would be definitely using a hash map and not a set. But I suspect that nestJS already provides mechanisms to get all clients in one room? Have you checked that?
2) It won't work on a multi container setup, but then again, I think nether the rest will. We would need a redis to share client status. => Could be actually a new issue, not but important right now.